indico/modules/events/registration/api.py

```diff
@@ -198,23 +147,10 @@ class RegistrantsHook(EventBaseHook):
 
     def _getParams(self):
         super(RegistrantsHook, self)._getParams()
-        self._conf_id = self._pathParams['event']
-        self._conf = ConferenceHolder().getById(self._conf_id)
+        self._event = Event.find_one(id=self._pathParams['event'])
 
     def _hasAccess(self, aw):
-        return self._conf.canManageRegistration(aw.getUser()) or self._conf.canModify(aw)
+        return self._event.can_manage(aw.getUser(), role='registration')
```

Comment:
Might be easier to change AccessWrapper.getUser like this:

    def getUser(self, legacy=True):
        return self._currentUser if legacy or self._currentUser is None else self._currentUser.user